lib/src/widgets/button/button.dart

```diff
@@ -77,12 +76,14 @@ class GazeButton extends StatelessWidget {
   final void Function()? onTap;
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
     final disabledColor = color == Colors.transparent ? color : color.withOpacity(0.3);
+    final volume = properties.withSound ? SoundVolume.getByNumber(ref.read(GazeInteractive().clickSoundVolume)) : SoundVolume.val0;
```

Comment:
Wenn hier `volume` und `type` mithilfe von riverpod nur rausgesucht werden, um sie dann in eine Methode einer Util Klasse reinzureichen und sonst nicht weiter im Code genutzt werden...

Könnte man dann nicht einen Provider machen, der sich `volume` und `type` raussucht und dann den Ton abspielt?
Damit würde die Util Klasse + Methode auch wegfallen, wenn man das alles in einem machen würde.

Also statt
`unawaited(_maybePlaySound(volume, type))` dann `ref.read(buttonMaybePlaySoundProvider)` 
(oder sowas Ähnliches)

Reply:
jap haste recht, da war der gedanke dass man den sound auch in den button reinreichen können soll von außen je button, aber das hab ich dann verworfen. dann mach ich da noch n provider hin :)